### ISSUE TITLE
Workspaces module crashes when moving between empty workspaces

### ISSUE THREAD

**Issue by jonvaldes** [Alexays/Waybar#283]
Repro steps:
- Have windows only in workspace 1.
- Switch to workspace 2 (empty)
- Switch to workspace 3 (empty)
- Switch back to workspace 2. Waybar will crash here with a segmentation fault.

Running waybar in gdb results in this stacktrace:
```
#0  0x00007ffff646e802 in _int_malloc (av=av@entry=0x7fffb8000020, bytes=bytes@entry=88) at malloc.c:3620
#1  0x00007ffff6470caa in __GI___libc_malloc (bytes=88) at malloc.c:3073
#2  0x00007ffff67f18f5 in operator new(unsigned long) () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff7d6e154 in std::_Rb_tree_node<std::pair<Json::Value::CZString const, Json::Value> >* std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_M_copy<std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_Alloc_node>(std::_Rb_tree_node<std::pair<Json::Value::CZString const, Json::Value> > const*, std::_Rb_tree_node_base*, std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_Alloc_node&) () at /usr/lib/x86_64-linux-gnu/libjsoncpp.so.1
#4  0x00007ffff7d6e1ad in std::_Rb_tree_node<std::pair<Json::Value::CZString const, Json::Value> >* std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_M_copy<std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_Alloc_node>(std::_Rb_tree_node<std::pair<Json::Value::CZString const, Json::Value> > const*, std::_Rb_tree_node_base*, std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_Alloc_node&) () at /usr/lib/x86_64-linux-gnu/libjsoncpp.so.1
#5  0x00007ffff7d6c51c in Json::Value::Value(Json::Value const&) () at /usr/lib/x86_64-linux-gnu/libjsoncpp.so.1
#6  0x00007ffff7d6e1ea in std::_Rb_tree_node<std::pair<Json::Value::CZString const, Json::Value> >* std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_M_copy<std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_Alloc_node>(std::_Rb_tree_node<std::pair<Json::Value::CZString const, Json::Value> > const*, std::_Rb_tree_node_base*, std::_Rb_tree<Json::Value::CZString, std::pair<Json::Value::CZString const, Json::Value>, std::_Select1st<std::pair<Json::Value::CZString const, Json::Value> >, std::less<Json::Value::CZString>, std::allocator<std::pair<Json::Value::CZString const, Json::Value> > >::_Alloc_node&) () at /usr/lib/x86_64-linux-gnu/libjsoncpp.so.1
#7  0x00007ffff7d6c51c in Json::Value::Value(Json::Value const&) () at /usr/lib/x86_64-linux-gnu/libjsoncpp.so.1
#8  0x00005555555ad0b6 in waybar::modules::sway::Workspaces::onCmd(waybar::modules::sway::Ipc::ipc_response) ()
#9  0x00005555555af2c6 in sigc::internal::slot_call<sigc::bound_mem_functor1<void, waybar::modules::sway::Workspaces, waybar::modules::sway::Ipc::ipc_response>, void, waybar::modules::sway::Ipc::ipc_response>::call_it(sigc::internal::slot_rep*, waybar::modules::sway::Ipc::ipc_response const&) ()
#10 0x00005555555a87ad in waybar::modules::sway::Ipc::sendCmd(unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) ()
#11 0x00005555555ac613 in waybar::modules::sway::Workspaces::onEvent(waybar::modules::sway::Ipc::ipc_response) ()
#12 0x00005555555af2c6 in sigc::internal::slot_call<sigc::bound_mem_functor1<void, waybar::modules::sway::Workspaces, waybar::modules::sway::Ipc::ipc_response>, void, waybar::modules::sway::Ipc::ipc_response>::call_it(sigc::internal::slot_rep*, waybar::modules::sway::Ipc::ipc_response const&) ()
#13 0x00005555555a8325 in waybar::modules::sway::Ipc::handleEvent() ()
#14 0x00005555555ac700 in std::_Function_handler<void (), waybar::modules::sway::Workspaces::worker()::{lambda()#1}>::_M_invoke(std::_Any_data const&) ()
#15 0x000055555558a329 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<waybar::util::SleeperThread::operator=(std::function<void ()>)::{lambda()#1}> > >::_M_run() ()
#16 0x00007ffff681c420 in  () at /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#17 0x00007ffff65cc182 in start_thread (arg=<optimized out>) at pthread_create.c:486
#18 0x00007ffff64f5b1f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95```

**Comment by Alexays** [comment]
I'm unable to reproduce the issue :/
Can you run gdb with debug symbols compiled?

**Comment by jonvaldes** [comment]
Gladly, although I'll be offline for a couple days, so I won't be able to do it until Thursday or so.

How would I build with debug symbols? This is the first time I've used meson/ninja, so not sure how to select the debug build

-------- Original Message --------

**Comment by Alexays** [comment]
Ok i got it, fixed with cccf60c30ec093362bc2c825dc8ccb9316d6bb32

**Comment by jonvaldes** [comment]
Bug fix confirmed :)
Loving waybar, thanks a lot for working on it!
